### ISSUE TITLE
Multiple pseudo counters with different dimensionality of the same physical counter fail to initialize after pool restart

### ISSUE THREAD

**Issue by tox1c90** [sardana-org/sardana#1438]
When I define multiple pseudo counters which act on the same physical 2D counter (image), but produce outputs with different number of dimensions, this only works as long as the Sardana pool is not restarted.

I have one pseudo counter integrating either over the whole image or a ROI, generating a 0D/scalar pseudo counter. Another pseudo counter integrates only along one axis, generating a 1D pseudo counter, the third one is just cutting out a part of the image or modifying the image and thus generates a 2D output.

I made an easy example of two pseudo counter controllers (0D, 2D) that reproduce the problem:

```
class dummy2DSumController(PseudoCounterController):

    counter_roles        = ('I',)
    pseudo_counter_roles = ('O',)
    
    def __init__(self, inst, props, *args, **kwargs):
        PseudoCounterController.__init__(self,inst,props, *args, **kwargs)
    
    def Calc(self, axis, counters):
        image = counters[0]        
        return image.sum()

class dummy2DInvertController(PseudoCounterController):

    counter_roles        = ('I',)
    pseudo_counter_roles = ('O',)
    
    def __init__(self, inst, props, *args, **kwargs):
        PseudoCounterController.__init__(self,inst,props, *args, **kwargs)
        
    def GetAxisAttributes(self, axis):
        axis_attrs = PseudoCounterController.GetAxisAttributes(self, axis)
        axis_attrs = dict(axis_attrs)
        axis_attrs['Value'][Type] = ((float, ), )
        axis_attrs['Value'][MaxDimSize] = (1024,1024)
        return axis_attrs
    
    def Calc(self, axis, counters):
        image = counters[0]        
        return -1*image
```

Now create a 2D dummy counter and set up both pseudo counters:

```
defctrl DummyTwoDController dummy2Dctrl
defelem dummy2D dummy2Dctrl

defctrl dummy2DSumController dummy2DSumCtrl I=dummy2D O=dummy2Dsum
defctrl dummy2DInvertController dummy2DInvertCtrl I=dummy2D O=dummy2Dinv
```

After adding both pseudo counters, `dummy2Dsum` and `dummy2Dinv` to the measurement group, they work properly and acquire data.
But when the Sardana pool is now restarted, it throws an error:

`MainThread     WARNING  2020-11-11 12:36:41,683 pc/dummy2dsumctrl/1: Failed to initialize dynamic attributes`

And the 0D pseudo counter is no longer operational:

`dummy2Dsum  =    <nodata>`

**Comment by reszelaz** [comment]
Hi Martin,
I have not yet tried the examples you have provided. From your detailed description to me it looks very related to the PR that @HEnquist is just working - #1441. I will start first from reviewing his PR and will then come back to this issue. If we are lucky then the issue may be already fixed.
Sorry for the inconveniences!

**Comment by reszelaz** [comment]
Hi again,
Sorry, I wrongly associated this bug report to #1441 but these two are not related.
I was able to reproduce your issue. It was solved by #1446 (the same PR that solved your other issue #1440) which is already intergrated in develop.
So, I will proceed to close this ticket. Simply use the latest develop branch or if you strictly need it we could hink on a hotfix release. But anyway in January we plan to release the [Jan21](https://github.com/sardana-org/sardana/milestone/8) milestone. 
Cheers!
